model/version.go

```diff
@@ -359,6 +359,15 @@ func GetVersionsWithOptions(projectId string, opts GetVersionsOptions) ([]Versio
 			"as":           "build_variants",
 		}
 		pipeline = append(pipeline, bson.M{"$lookup": lookupBuilds})
+
+		if opts.ByBuildVariant != "" {
```

Comment:
I believe including this with the very first match will perform better

Reply:
there's a build_variants_status.activated field. Or was this the not wanting to use the build cache in new code?

Reply:
I looked into this with some existing versions; I don't think it's used for all builds, it looks like it's more for batchtime builds.